### PR TITLE
feat/APPT_STQA_145_View_Appointments_Cancel_And_Filter

### PR DESCRIPTION
**Pull request by Rakaesa**
Cancel functionality and unit tests.

### CODE REVIEW

File: src/main/webapp/WEB-INF/jsp/appointments/viewAppointments.jsp [hunk 1]
```diff
@@ -20,16 +27,32 @@
...
             <td>${v.pet.name}</td>
             <td>${v.description}</td>
+            <td><input type="submit" id=${v.id} value="Cancel"></td>
         </tr>
+        </form>
     </c:forEach>
     </table>
 </body>
 </petclinic:layout>
+<%!
```
Comment (sylveon) [inline src/main/webapp/WEB-INF/jsp/appointments/viewAppointments.jsp]: Remove this

File: src/test/java/org/springframework/samples/petclinic/service/AbstractClinicServiceTests.java [hunk 2]
```diff
@@ -366,12 +368,29 @@ void shouldDeleteVisitsById() {
...
 
         Collection<Visit> visits = this.clinicService.findAllVisits();
         assertThat(visits.size()==4);
 
     }
 
+    @Test
+    @Transactional
+    @Order(19)
+    void testCancelFunctionality() throws Exception{
```
Comment (sylveon) [inline src/test/java/org/springframework/samples/petclinic/service/AbstractClinicServiceTests.java]: This should follow the same naming convention as the other tests (rename to `shouldDeleteVisitsById` and rename the existing `shouldDeleteVisitsById` to `shouldDeleteVisitsByIdIn`)

**Review by sylveon — CHANGES_REQUESTED** [review]

**Review by GSwizzy — APPROVED** [review]

**Review by GSwizzy — DISMISSED** [review]
Don't forget to rename your PR 👍

**Review by sylveon — APPROVED** [review]

File: src/test/java/org/springframework/samples/petclinic/service/AbstractClinicServiceTests.java (hunk 2 above)
Comment (Rakaesa) [reply src/test/java/org/springframework/samples/petclinic/service/AbstractClinicServiceTests.java]: Added requested changes in latest push.

**Review by GSwizzy — APPROVED** [review]

**Pull request merged**